src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const groupedByKey:GroupsMap<T> = {};
+
+  items.forEach((item: T) => {
+    return (groupedByKey[`${item[key]}`] = groupedByKey[`${item[key]}`]
+    || []).push(item);
+  });
```

Comment:
Why do you return something from the callback on forEach? For what? You need just update the groupedByKey, it's all